### PR TITLE
Port Solidity deposit contract - r2

### PR DESCRIPTION
**Pull request by hwwhww**
Address #1968 
Replace #1987 

- [x] Port the whole repo with commit history
- [x] Integrate the dapp.tools tests with CI
- [x] Clean up the configuration and add the web3.py tester back to the CI
- [x] Clean up the documents

### CODE REVIEW

File: solidity_deposit_contract/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# eth2-deposit-contract
```
Comment (hwwhww) [inline solidity_deposit_contract/README.md:1]: /cc @axic 
I added a note of `r2`. Let me know if you want to add something. :)

File: solidity_deposit_contract/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+## History
+
+This is a rewrite of the [Vyper Eth 2.0 deposit contract](https://github.com/ethereum/eth2.0-specs/blob/v0.12.2/deposit_contract/contracts/validator_registration.vy) to Solidity.
+
+The original motivation was to run the SMTChecker and the new Yul IR generator option (`--ir`) in the compiler.
+
+As of June 2020, this contract (the version tagged as [`r1`](https://github.com/axic/eth2-deposit-contract/tree/r1) in GitHub repository [axic/eth2-deposit-contract](https://github.com/axic/eth2-deposit-contract/tree/r1)) has been verified and is considered for adoption.
```
Comment (djrtwo) [inline solidity_deposit_contract/README.md:9]: Linking to the external axic repo might be confusing to leave in the README. I would consider a rewrite to be spec-repo centric or at least mention the migration and deprecation of the original repo

**Comment by MrChico** [comment]
Proofs for the new bytecode are being rerun here: https://github.com/runtimeverification/deposit-contract-verification/pull/17

**Comment by daejunpark** [comment]
Rerunning proofs should be done in less than an hour if succeeds.

**Comment by daejunpark** [comment]
Rerunning has been successfully finished! https://office.runtimeverification.com/jenkins/blue/organizations/jenkins/deposit-contract-verification/detail/PR-17/1/pipeline

**Comment by djrtwo** [comment]
Excellent! Thank you @daejunpark :)

**Comment by djrtwo** [comment]
any blocker on this @hwwhww ?

File: solidity_deposit_contract/README.md (hunk 2 above)
Comment (hwwhww) [reply solidity_deposit_contract/README.md]: I just rewrote it a bit. post-merge revise is welcome!

**Comment by axic** [comment]
For reference this is an import of https://github.com/axic/eth2-deposit-contract/releases/tag/r2.

**Review by cornbread78 — APPROVED** [review]

**Review by Allzz77123 — APPROVED** [review]

**Pull request merged**